### PR TITLE
chore: simp tracing reports ←

### LINKED ISSUE #2131: Incorrect simp tracing for local right-to-left simp theorems (`[<- x]`)
MWE:

```lean
structure A
structure B

theorem stuff : A = B := sorry

example : B := by
  set_option tactic.simp.trace true in
  simp [← stuff] -- Try this: simp only [stuff] // should be: simp only [← stuff]
  exact A.mk
```

Tested with Lean 4 nightly 2023-03-01.

Reported on Zulip: https://leanprover.zulipchat.com/#narrow/stream/270676-lean4/topic/.60simp.3F.60.20dropping.20.60.3C-.60

### PR DESCRIPTION
**Pull request by kim-em**
Fixes #2131 and https://github.com/leanprover/std4/issues/264

### CODE REVIEW

**Comment by semorrison** [comment]
@digama0 if it's useful for reviewing I can make proposed downstream changes to Std and Mathlib as well, but won't immediately unless you say so.

**Comment by semorrison** [comment]
@digama0, I'd like to get this in, but am uncertain whether your self-assignment above means:
* that I shouldn't do further work on this because you have in mind a better solution?
* that I should do the downstream fixes to Std and Mathlib required for this approach, and then you'll review?

**Comment by digama0** [comment]
It means I was planning on reviewing it, but I was at a conference last week so I'm just now working down my backlog. I'll take a look at it now or tomorrow.

File: src/Lean/Meta/Tactic/Simp/SimpTheorems.lean [hunk 1]
```diff
@@ -18,7 +18,7 @@ what action the user took which lead to this theorem existing in the simp set.
 -/
 inductive Origin where
   /-- A global declaration in the environment. -/
-  | decl (declName : Name)
+  | decl (declName : Name) (inv : Bool)
```
Comment (digama0) [inline src/Lean/Meta/Tactic/Simp/SimpTheorems.lean:21]:
```suggestion
  | decl (declName : Name) (inv := false)
```

File: src/Lean/Meta/Tactic/Simp/SimpTheorems.lean [hunk 2]
```diff
@@ -42,9 +42,9 @@ inductive Origin where
 
 /-- A unique identifier corresponding to the origin. -/
 def Origin.key : Origin → Name
-  | .decl declName => declName
+  | .decl declName _ => declName
   | .fvar fvarId => fvarId.name
-  | .stx id _ => id
+  | .stx id _  => id
```
Comment (digama0) [inline src/Lean/Meta/Tactic/Simp/SimpTheorems.lean:47]:
```suggestion
  | .stx id _ => id
```

**Review by digama0 — APPROVED** [review]

**Comment by semorrison** [comment]
Phew, lots of downstream changes required for this one:
* https://github.com/JLimperg/aesop/pull/72
* https://github.com/leanprover/std4/pull/284

**Comment by semorrison** [comment]
I've checked that downstream projects survive okay. There is some unfortunate copypasta there that ideally we can clean up later.

**Review by digama0 — APPROVED** [review]

**Pull request merged**